api/src/main/java/org/commonjava/indy/util/LocationUtils.java

```diff
@@ -50,10 +52,15 @@ public static KeyedLocation toLocation( final ArtifactStore store )
             return null;
         }
 
+        KeyedLocation location = (KeyedLocation) store.getTransientMetadata( KEYED_LOCATION_METADATA );
+        if ( location != null )
+        {
+            return location;
+        }
+
```

Comment:
If one hosted re-stored with some attributes relevant with location, like 'readonly', this KEYED_LOCATION_METADATA retrive logic may keep the location with old(original) one, will not update its isReadOnly value. Maybe open the set method for the allowsDeletion is a fast way to resolve? (I'm not sure if it's a good solution, current CI failure should be caused by this I suppose)

Reply:
This is exactly what's happening in the master and 1.2.x branches. I've introduced a listener there to clear the stored Location object when the store is updated (in those branches) but need to cherry-pick it here still.